lib/policies/load-balancing.js

```diff
@@ -294,15 +297,28 @@ TokenAwarePolicy.prototype.newQueryPlan = function (keyspace, queryOptions, call
  */
 function TokenAwareIterator(keyspace, queryOptions, replicas, childPolicy) {
   this.keyspace = keyspace;
-  this.replicas = replicas;
   this.childPolicy = childPolicy;
   this.queryOptions = queryOptions;
-  this.childHosts = [];
-  this.remoteReplicas = [];
+  this.localReplicas = [];
+  this.localReplicaIndex = 0;
+  this.remoteReplicaIndex = 0;
```

Comment:
Excellent, thanks!  This looks great.  I have a few test updates coming very shortly, will be +1ing the PR soon.